### PR TITLE
Add basic documentation for the FFI in Effekt

### PR DESCRIPTION
**Pull request by matthias-dunkel**

### CODE REVIEW

File: docs/concepts/FFI.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+layout: docs
+title: Foreign Function Interface
+permalink: docs/concepts/ffi
+---
+# Foreign Function Interface
+Effekt implements a Foreign Function Interface (FFI) for the host languages JavaScript and Java. With Effekt’s FFI it is possible to use the host language functions, to model hosts objects, and to use their methods. The FFI differentiates between value types, which do not have to be tracked by Effekt for any changes, and are opaque (Effekt does not understand the semantics of the FFI types), and interface types that describe objects that need to be tracked by Effekt, and can be used to model FFI capabilities. To track any interfaces, one has to define resources which should be tracked by Effekt and let the object live in these resources.
```
Comment (b-studios) [inline docs/concepts/FFI.md:7]:
```suggestion
Effekt implements a foreign function interface (FFI) for the several host languages (JavaScript, ChezScheme, LLVM, SML, etc.). With Effekt’s FFI it is possible to use the host language functions, to model hosts objects, and to use their methods. 

Like the Effekt language, the FFI differentiates between _value types_ and _interface types_, both of which are opaque in a sense that Effekt does not make any assumptions about their semantics.

_Value types_ do not have to be tracked by Effekt and whereas _interface types_ need to be tracked by Effekt.  Interface types often describe objects or resources and can be used to model FFI capabilities. To track any interfaces, one has to define resources which should be tracked by Effekt and let the object live in these resources (see below).
```

File: docs/concepts/FFI.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Using external functions with value types:
+Let's assume we want to use JavaScript's `Math.random` function in Effekt.
+```effekt
+extern io def random(): Double = "Math.random()"
+```
+The defintion of an external function is indicated by the keyword `extern` followed by `io`. `io` tells the compiler, that the external function uses the `io` resource, which is tracked by Effekt. The Effekt compiler can not check to typesafety, so the return type annotation needs to ensured by the programmer.
```
Comment (b-studios) [inline docs/concepts/FFI.md:14]: Maybe you could explain that `io` is really syntactic sugar for the singleton set `{io}` and `io` is a builtin resource. In general arbitrary resources can be annotated there. Similarly, `pure` is "sugar" for `{}`, and `control` is a special resource used to mark builtins that have access to the continuation (we need a whole section to explain that :) ).

File: docs/concepts/FFI.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Interfaces and resources
+Interfaces can be used to model external objects. These objects need to be tracked by Effekt with a resource. 
+```effekt
+extern interface Array
+extern resource stack: Array
+extern {stack} def makeArray(size: Int): Array at {stack} = "size"
+```
+The definition of an external function is indicated by the keyword `extern` followed by `io`. `io` tells the compiler, that the external function uses the `io` resource, which is tracked by Effekt. The Effekt compiler can not check for type safety, so the return type annotation needs to be ensured by the programmer.
```
Comment (b-studios) [inline docs/concepts/FFI.md:23]: Here you talk about `io` but the example uses `stack`.

**Review by b-studios — CHANGES_REQUESTED** [review]
Hey, thanks for the PR which is very much appreciated! I have a few comments below.

**Comment by b-studios** [comment]
Hey, do you still plan to contribute this PR?

**Comment by matthias-dunkel** [comment]
Yes, sorry. The last months were pretty full. I will find some time to contribute in the next week!

**Comment by b-studios** [comment]
Don't worry, just wanted to check whether I should choose the pr :)

**Comment by jiribenes** [comment]
Closed for inactivity, a quick intro to current FFI is now in https://github.com/effekt-lang/effekt/pull/536. (cc @dvdvgt)

**Pull request closed without merging**